src/bindings/js/node/tests/infer_request.test.js

```diff
@@ -193,3 +194,14 @@ describe('InferRequest', () => {
     assert.deepStrictEqual(res1['fc_out'].data[0], res2['fc_out'].data[0]);
   });
 });
+
+describe("Asynchronous inference", () => {
+    it('Test promisifiedAsyncInfer()', () => {
+        const ir = compiledModel.createInferRequest();
+        const res1 = ir.infer([tensorData]);
+        const promisifiedAsyncInfer = util.promisify(ov.asyncInfer);
+        promisifiedAsyncInfer(ir, [tensorData]).then((result, err) => {
+            assert.deepStrictEqual(res1['fc_out'].data[0], result['fc_out'].data[0]);
+        })
+     });
```

Comment:
1. I propose not to use promisify, let's use callback as is.
2. Please, pay attention that callback function must have error as the first parameter and rest as data.

```suggestion
    it('Compare infer and asyncInfer results', () => {
        const ir = compiledModel.createInferRequest();
        const res1 = ir.infer([tensorData]);

        ov.asyncInfer(ir, [tensorData], (err, result) => {
            assert.deepStrictEqual(res1['fc_out'].data[0], result['fc_out'].data[0]);
        });
     });
```

Reply:
Here I made a mistake in my comment. `then` function [takes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then#parameters) two functions as parameters. You have one function and this function should have only one parameter. In your case it is `result`. `err` parameter in this case is redundant and always be equal `undefined`.